pycloudlib/azure/cloud.py

```diff
@@ -615,15 +632,24 @@ def _create_virtual_machine(
         if vm_params:
             update_nested(params, vm_params)
         self._log.debug("Creating Azure virtual machine: %s", name)
-        vm_poller = (
-            self.compute_client.virtual_machines.begin_create_or_update(
-                self.resource_group.name,
-                name,
-                params,
+        try:
+            vm_poller = (
+                self.compute_client.virtual_machines.begin_create_or_update(
+                    self.resource_group.name,
+                    name,
+                    params,
+                )
+            )
+            vm_poller.wait(300)
```

Comment:
@jrtknauer Do you think we could have a timeout here, considering that it seems Azure's timeout is like 20 mins. I found out this default timeout from testing failed images. I am defaulting to 5 mins since I notice most of successful provisioning have taken like 3 mins.

Reply:
I would default to the Azure timeout and ideally provide an option for a user to override. Burying a timeout - especially when its explicitly truncating the Azure default could end up backfiring on users if there's an unexpectedly slow deployment.